Adding an article regarding Python binding locator changes

joshmgrant opened the pull request:
In an upcoming version of the Python bindings, some locator-specific
methods are being removed, which will likely cause a lot of
Python binding users pain and frustration. This post tries to
help make people aware of these changes.

**Thanks for contributing to the Selenium site and documentation!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.


<!--- Provide a general summary of your changes in the Title above -->

### Description
I'm adding a new entry to the Selenium blog! It's about changes to the Python bindings regarding find_element_* methods.

### Motivation and Context
Removing the find_element_by* may cause some confusion for Python users of Selenium, so I tried to help with a blog entry about how to update locators for these new changes.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [ ] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.
<!--- Provide a general summary of your changes in the Title above -->

joshmgrant committed:
Adding an article regarding Python binding locator changes

In an upcoming version of the Python bindings, some locator-specific
methods are being removed, which will likely cause a lot of
Python binding users pain and frustration. This post tries to
help make people aware of these changes.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/seleniumhq.github.io?pullRequest=945) <br/>All committers have signed the CLA.

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
```diff
@@ -0,0 +1,59 @@
+---
+title: "Locate Your Locators - Python Bindings Changes Upcoming"
+linkTitle: "Locate Your Locators Python Bindings Changes Upcoming"
```
Could you please modify the title so it follows our guideline?
https://www.selenium.dev/documentation/about/style/#capitalization-of-titles

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
```diff
@@ -0,0 +1,59 @@
+categories: ["general"]
+author: Josh Grant ([@joshin4colours](https://twitter.com/joshin4colours))
+description: >
+  Updating Your Locators for Changes in the Pythom Bindings in Selenium 4
```
Same as above
https://www.selenium.dev/documentation/about/style/#capitalization-of-titles

diemol replied:
Well, this is not a title, but I hope you get what I mean.

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
```diff
@@ -0,0 +1,59 @@
+  Updating Your Locators for Changes in the Pythom Bindings in Selenium 4
+---
+
+(This article was originally posted [here](https://simplythetest.tumblr.com/post/674917293614039040/locate-your-locators-python-bindings-changes). Thanks to the Selenium core contributors for adding this here!)
+
+In real estate, the mantra for finding a new house or office space is "location, location, location!". It could be said that when working with Selenium, a critical aspect of writing tests is "locators, locators, locators!". Having a robust locator strategy - in your app under test and in your test framework - is highly important for effective testing.
+
+If you are a Pythonista like myself and using Selenium for your test automation, then there are some important changes coming to how locators are defined and used.
```
Could you please break the lines to about 100 chars per line?
https://www.selenium.dev/documentation/about/style/#line-length

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
```diff
@@ -0,0 +1,59 @@
+
+If you are a Pythonista like myself and using Selenium for your test automation, then there are some important changes coming to how locators are defined and used.
+
+As of Selenium 4.1.2, the Python Selenium bindings will remove locator-specific methods for finding elements. This means that the methods
```
Wasn't it 4.3.0?

![image](https://user-images.githubusercontent.com/5992658/152044747-16a482dc-6126-4230-a4c9-a84ae13d1d6e.png)

joshmgrant replied:
Interesting, this looks like an old version of my article, not sure how that got in there. Good catch!

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+
+As of Selenium 4.1.2, the Python Selenium bindings will remove locator-specific methods for finding elements. This means that the methods
+
+```
````
```suggestion
```python
```

diemol replied:
So the syntax highlighting looks nicer.

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+```
+will be removed. All of these methods are in fact special cases of
+
+```
````
```suggestion
```python
```

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+
+Note that it's good practice to use the [By object](https://www.selenium.dev/selenium/docs/api/py/webdriver/selenium.webdriver.common.by.html#module-selenium.webdriver.common.by) which has specific values for using particular locator strategies. For example, this line
+
+```
````
```suggestion
```python
```

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+
+becomes
+
+```
````
```suggestion
```python
```

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+
+If you're really desperate however you can use strings instead of the By object:
+
+```
````
```suggestion
```python
```

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
````diff
@@ -0,0 +1,59 @@
+driver.find_element('css selector', '.myelement child').text
+```
+
+If you have any plans to upgrade your Selenium client for your Python tests to recent versions of Selenium 4, definitely keep these changes in mind. It's a good time to update your locator strategy and structure.
````
Line lenght to around 100 chars, please :)

diemol commented on website_and_docs/content/blog/2022/python_locators_se4.md:
```diff
@@ -0,0 +1,59 @@
+  Updating Your Locators for Changes in the Pythom Bindings in Selenium 4
+---
+
+(This article was originally posted [here](https://simplythetest.tumblr.com/post/674917293614039040/locate-your-locators-python-bindings-changes). Thanks to the Selenium core contributors for adding this here!)
```
Maybe this line can be at the bottom of the post?

diemol reviewed (changes requested):
Thank you, @joshmgrant! Looks good, a few minor changes requested.

joshmgrant committed:
Some updates based on feedback from @diemol

Changes were made based mostly on formatting and
style guides, including using Python blocks
for code, restricting lines to 100 characters
and formatting titles properly.

diemol reviewed (approved):
Thank you, @joshmgrant!

Looking forward to more contributions!
